geant4.sh

```diff
@@ -33,6 +33,12 @@ env:
 # if this variable is not defined default it to OFF
 : ${GEANT4_BUILD_MULTITHREADED:=OFF}
 
+# We embed Xerces-C directly into GEANT4 to avoid relocation issues and
+# incompatibility with other possible instances of XercesC
+if [ ! "X$XERCES_ROOT" = X ]; then
+  rsync -av $XERCES_ROOT/ $INSTALL_ROOT/
+fi
```

Comment:
I don't understand why this is necessary. It seems like another hack. Can't we use XercesC just as any other package/library? (Your previous fix was much shorter)

Reply:
The previous commit does not actually fully work, given CMake does not expand environment variables. It was only ok for `geant4-config`.

The problem is that Geant4 embeds the absolute path to the Xerces installation in some of its scripts, which then cannot be automatically relocated (because the Geant4 root is different from the Xerces root). This is actually the only package which does it, i.e. the only one which pretends to know where its dependencies are installed, rather than delegating it  to the dependency itself. A proper solution would be to generate a CMake config file for XercesC and use that rather than `geant4-config` or the Cmake hardcoded value. An alternative would be to patch by hand all the occurrences of XercesC in Geant4 and use a relative path.

This solution, i.e. bundling xerces inside geant4, (once I managed to get it to work) is probably the cleanest and will also have the advantage to avoid compatibility issues.

Reply:
Ok. But can't we directly refer to the final XercesC installation (after relocation) when building Geant4? Why did it work on the CI?